src/shared/sleep-config.c

```diff
@@ -802,6 +802,12 @@ static bool location_is_resume_device(const HibernateLocation *location, dev_t s
  * Attempt to find the hibernation location by parsing /proc/swaps, /sys/power/resume, and
  * /sys/power/resume_offset.
  *
+ * Beware:
+ *  Never use a device or file as location that hasn't been either explicitly specified by the root user (for example via
+ *  /sys/power/resume) or that isn't an already active swap area!
+ *  Otherwise various security attacks might become possible, for example an attacker could silently attach such a device
+ *  and circumvent full disk encryption when it would be automatically used for hibernation.
```

Comment:
i don't find that logic convincing. we only do auto-discovery and use of partitions on the root fs anyway, and that's not going to change. Anything else would be insecure. Moreover, even if we'd add something like what you describe (i.e. automatically use swap partitions on arbitrary block devices), then this is still riskless – as long as we do so with encryption.

Hence. Not convinced we need to add this here. The hinted case (auto-use of arbitrary swap devices without encryption) is bogus, we don't do that, and won't do that.

Reply:
@poettering Well, I don't think it makes a difference where (like e.g. "**only** on the block device backing the root fs anyway") auto-discovery is made. That block device could simply have had enough space left, so that an attacker could (offline) silently add a large enough (and unencrypted) swap device.<br>
As for your second argument:<br>
I think it's all to easy to simply forget about the need for doing this only with encryption in place. As mentioned above, the `TODO` even describes a feature that would likely allow abuse, without any mentioning that this must happen only when on top of encryption.

But sure, if you're certain that if any of this is ever implemented it will not be forgotten to do it in a secure manner it's not needed - I just felt, better safe than sorry.